src/notification.c

```diff
@@ -270,38 +269,62 @@ notification_elem_end(void *data, const char *el)
 	}
 }
 
+/* XXXCJ this needs more cleanup and error checking */
 void
 save_notification_data(struct xmldata *xml_data)
 {
-	char *notification_filename = generate_filename_from_uri(xml_data->uri, xml_data->opts->basedir_primary, "https://");
+	char *notification_filename;
+
+	if (asprintf(&notification_filename, "%s/.state",
```

Comment:
Hmmm will need to consider the ".state" file further in the future both of these were already issues:
- what happens if they publish a delta file called "<host>/dir/.state" (we discussed this early on)
- how should we handle the same host with a new notification.xml location (is it the "same") ie. can a single host have multiple notification streams

Reply:
Yes. I checked routinator and it looks like it uses /dir/state.txt as filename. The second point is also interesting since the assumption is that there is only one RRDP repo per host (we use the rrdp hostname as base name). I seriously hope that RRDP server operators change the session UUID if they modify streams like this.